### PR TITLE
Viite 426 check if new roadpart is already reserved

### PR DESCRIPTION
**Pull request by Campanha**
Check in project_link table for same roadNumber & roadPart but ignoring current project id
Show popup in UI after error in saving new road address in project

### CODE REVIEW

File: digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/dao/ProjectDAO.scala [hunk 1]
```diff
@@ -173,6 +173,19 @@ object ProjectDAO {
     Q.queryNA[String](query).firstOption
   }
 
+
+  def isNewRoadPartLinkUsed(roadNumber: Long, roadPartNumber: Long, projectId: Long) = {
+    val query =
+      s"""
+		select prol.id
+         from project_link prol
+         join lrm_position pos on prol.lrm_position_id = pos.id
```
Comment (oappicgi) [inline digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/dao/ProjectDAO.scala]: where do we need lrm_position?

File: digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/ProjectService.scala [hunk 2]
```diff
@@ -60,8 +60,8 @@ class ProjectService(roadAddressService: RoadAddressService, roadLinkService: Ro
     * @return
     */
   def checkNewRoadAddressNumberAndPart(roadNumber: Long, roadPart: Long, project :RoadAddressProject): Option[String] = {
-      val roadAddresses = RoadAddressDAO.isNewRoadPartUsed(roadNumber, roadPart, project.id)
-      if (roadAddresses.isEmpty) {
+      val projectLinks = ProjectDAO.isNewRoadPartLinkUsed(roadNumber, roadPart, project.id)
```
Comment (oappicgi) [inline digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/ProjectService.scala]: we also need to keep old check

File: digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/dao/ProjectDAO.scala (hunk 1 above)
Comment (pedrosag) [inline digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/dao/ProjectDAO.scala]: Possibly to make sure that the Project_Link has a LRM_Possition connection therefore being a step closer to prove that it is correctly created (therefore being able to be used)?

Comment (Campanha) [inline digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/dao/ProjectDAO.scala]: New version commit will not have this query. As Ossi said is more deeper than that, we need to keep the validation of:
1.new road address should not exist already in Road_address table
2.new road_address (same roadNumber and roadPart) should not exist in Project_link for others project ids

**Pull request merged**